packages/docusaurus-plugin-content-docs/src/routes.ts

```diff
@@ -253,3 +277,22 @@ export async function createAllRoutes(
   const routes = await buildAllRoutes(param);
   routes.forEach(param.actions.addRoute);
 }
+
+function compositeLastModified<T>(
+  iterable: T[],
+  mapToLastModified: (item: T) => Date | undefined,
+): Date | undefined {
+  const lastModifieds = iterable.map(mapToLastModified);
+  let max;
+  // we only want a last modified date,
+  // if we can say with certainty that all changes are accounted.
+  for (const lastModified of lastModifieds) {
+    if (lastModified === undefined) {
+      return undefined;
+    }
+    if (max === undefined || lastModified > max) {
+      max = lastModified;
+    }
+  }
+  return max;
+}
```

Comment:
https://developers.google.com/search/blog/2023/06/sitemaps-lastmod-ping#the-lastmod-element

> You can use a lastmod element for all the pages in your sitemap, or just the ones you're confident about. For instance, some site software may not be able to easily tell the last modification date of the homepage or a category page because it just aggregates the other pages on the site. In these cases it's fine to leave out lastmod for those pages.

I think we don't need this, and can only focus on adding the lastmod attribute for pages actually backed by a single markdown document.